core/utils.go

```diff
@@ -301,3 +303,37 @@ func FileExists(path string) (bool, error) {
 
 	return false, err
 }
+
+// get body text with a text editor
+// works with terminal based editors
+func getBodyTextWithEditor(editor string) (string, error) {
+	bodyTxtFile, err := os.CreateTemp(os.TempDir(), "changie-body-txt-")
+	if err != nil {
+		return "", err
+	}
+	defer bodyTxtFile.Close()
+	defer os.Remove(bodyTxtFile.Name())
+
+	if editor == "" {
+		editor = "vim"
+	}
+
+	cmd := exec.Command("sh", "-c", editor+" "+bodyTxtFile.Name())
+
+	// Set the stdin and stdout of the command to the current process's stdin and stdout
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = os.Stdout
+
+	if err = cmd.Run(); err != nil {
+		return "", fmt.Errorf("error opening the editor: %v", err)
+	}
+
+	buf, err := io.ReadAll(bodyTxtFile)
+	if err != nil {
+		if err != io.EOF {
```

Comment:
I don't believe ReadAll will ever return an EOF error.

Reply:
My bad. I will fix it

Reply:
I find it best if I close the comments as I see them resolved, just makes it easier to keep track as a reviewer.